packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { useState } from 'react';
+import styled, { css, keyframes } from 'styled-components';
+import { FloatingFocusManager, useFloating } from '@floating-ui/react';
+import usePresence from 'use-presence';
+import { ModalPortal } from './ModalPortal';
+import { ModalHeader } from './ModalHeader';
+import type { ModalHeaderProps } from './ModalHeader';
+import { ModalFooter } from './ModalFooter';
+import type { ModalFooterProps } from './ModalFooter';
+import { ModalBody } from './ModalBody';
+import type { ModalBodyProps } from './ModalBody';
+import { ModalContext } from './ModalContext';
+import { ModalBackdrop } from './ModalBackdrop';
+import {
+  modalBorderRadius,
+  modalMaxHeight,
+  modalMaxWidth,
+  modalMinWidth,
+  modalResponsiveScreenGap,
+} from './modalTokens';
+import {
+  MetaConstants,
+  castWebType,
+  isValidAllowedChildren,
+  makeAccessible,
+  makeMotionTime,
+  makeSize,
+  metaAttribute,
+} from '~utils';
+import { BaseBox } from '~components/Box/BaseBox';
+import { useTheme } from '~components/BladeProvider';
+import { Box } from '~components/Box';
+
+const entry = keyframes`
+  from {
+    opacity: 0;
+    transform: translate(-50%, -50%) scale(0.9) translateY(20px);
+  }
+  to {
+    opacity: 1;
+    transform: translate(-50%, -50%) scale(1) translateY(0px);
+  }
+`;
+
+const exit = keyframes`
+  from {
+    opacity: 1;
+    transform: translate(-50%, -50%) scale(1) translateY(0px);
+  }
+  to {
+    opacity: 0;
+    transform: translate(-50%, -50%) scale(0.9) translateY(20px);
+  }
+`;
+
+const ModalContent = styled(BaseBox)<{ isVisible: boolean }>(({ isVisible, theme }) => {
+  return css`
+    box-shadow: ${theme.elevation.highRaised};
+    position: fixed;
+    transform: translate(-50%, -50%);
+    opacity: ${isVisible ? 1 : 0};
+    animation: ${isVisible ? entry : exit}
+      ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+      ${
+        isVisible
+          ? castWebType(theme.motion.easing.entrance.revealing)
+          : castWebType(theme.motion.easing.exit.revealing)
+      }};
+  `;
+});
+
+type ModalProps = {
+  children: React.ReactNode;
+  isOpen: boolean;
+  onDismiss: () => void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  initialFocusRef?: React.MutableRefObject<any>;
+  size?: 'small' | 'medium' | 'large';
+  accessibilityLabel?: string;
+};
+
+const Modal = ({
+  isOpen = false,
+  children,
+  onDismiss,
+  initialFocusRef,
+  size = 'small',
+  accessibilityLabel,
+}: ModalProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [footerHeight, setFooterHeight] = useState(0);
```

Comment:
Need footer height to calculate scroll overlay offset. Added a comment about it in footer